crates/fiber-lib/src/cch/actor.rs

```diff
@@ -440,39 +439,25 @@ impl CchActor {
             Ok(order) => order,
         };
 
+        let hash = Hash256::from_str(&event.payment_hash)?;
+
         order.status = event.status;
-        if let (Some(preimage), Some(network_actor), Some(channel_id), Some(tlc_id)) = (
-            event.preimage,
-            &self.network_actor,
-            order.channel_id,
-            order.tlc_id,
-        ) {
+        if let Some(preimage_str) = event.preimage {
```

Comment:
do we need to verify whether it's match for `payment_hash` and `preimage` here?

Reply:
Network actor will return the error